Content.Server/Silicons/Bots/MedibotSystem.cs

```diff
@@ -0,0 +1,175 @@
+using Content.Server.NPC.Components;
+using Content.Server.Chemistry.EntitySystems;
+using Robust.Shared.Physics.Components;
+using Robust.Server.GameObjects;
+
+namespace Content.Server.Silicons.Bots
+{
+    public sealed class MedibotSystem : EntitySystem
+    {
+        [Dependency] private readonly MobStateSystem _mobs = default!;
+        [Dependency] private readonly SolutionContainerSystem _solution = default!;
+        [Dependency] private readonly PopupSystem _popups = default!;
+        [Dependency] private readonly DoAfterSystem _doAfter = default!;
+        [Dependency] private readonly AudioSystem _audioSystem = default!;
+        [Dependency] private readonly ActionBlockerSystem _blocker = default!;
+        [Dependency] private readonly PhysicsSystem _physics = default!;
+        [Dependency] private readonly SharedInteractionSystem _interactionSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<MedibotComponent, InteractNoHandEvent>(PlayerInject);
+            SubscribeLocalEvent<MedibotComponent, DoAfterEvent<MedibotInjectData>>(OnDoAfter);
+        }
+
+        /// Yeah it's not super ideal that players have a different code path to start injecting,
+        /// but they have parity in being able to do it now unlike before.
+        private void PlayerInject(EntityUid uid, MedibotComponent component, InteractNoHandEvent args)
+        {
+            if (args.Target == null)
+                return;
+
+            if (args.Target == uid)
+                return;
+
+            if (!SharedInjectChecks(uid, args.Target.Value, out var injectable))
+                return;
+
+            TryStartInject(uid, component, args.Target.Value, injectable);
+        }
+        public bool NPCStartInject(EntityUid uid, EntityUid target, MedibotComponent? component = null)
+        {
+            if (!Resolve(uid, ref component))
+                return false;
+
+            if (!SharedInjectChecks(uid, target, out var injectable))
+                return false;
+
+            return TryStartInject(uid, component, target, injectable);
+        }
+
+        private void OnDoAfter(EntityUid uid, MedibotComponent component, DoAfterEvent<MedibotInjectData> args)
+        {
+            component.IsInjecting = false;
+
+            if (args.Cancelled || args.Handled || args.Args.Target == null)
+                return;
+
+            _audioSystem.PlayPvs(component.InjectFinishSound, args.Args.Target.Value);
+            _solution.TryAddReagent(args.Args.Target.Value, args.AdditionalData.Solution, args.AdditionalData.Drug, args.AdditionalData.Amount, out var acceptedQuantity);
+            _popups.PopupEntity(Loc.GetString("hypospray-component-feel-prick-message"), args.Args.Target.Value, args.Args.Target.Value);
+            EnsureComp<NPCRecentlyInjectedComponent>(args.Args.Target.Value);
+            args.Handled = true;
+        }
+
+
```

Comment:
Clean up your whitespace and formatting. If you're using Rider, use the autoformatter.